base/server/python/pki/server/deployment/__init__.py

```diff
@@ -866,7 +869,7 @@ def import_cert_request(self, subsystem, tag, request):
             adjust_validity=request.systemCert.adjustValidity)
 
         if request_id_generator == 'random':
```

Comment:
I think this should `!= legacy`. If in the future we will add additional methods then they should be managed like the _random_ case in the java code and here you get the value in any case

Reply:
Good point! I updated the code.